tests/test_datasets.py

```diff
@@ -0,0 +1,41 @@
+import yaml
+import urllib.request
+import pytest
+
+
+d = yaml.load(open("mir-datasets.yaml"))
+
+required_fields = [
+    'url',
+    'metadata',
+    'contents',
+    'audio'
+]
+
+def datasets():
+    for name, values in d.items():
+        yield name, values
+
+
+@pytest.fixture(
+    params=datasets(),
+    ids=['%s' % k for k, l in datasets()]
+)
+def dataset(request):
+    return request.param
+
+
+def test_datasets(dataset):
+    name = dataset[0]
+    values = dataset[1]
+
+    # check for spaces in beginning or end of name
+    assert ' ' not in [name[0], name[-1]]
+
+    # check if website is up
+    code = urllib.request.urlopen(values['url'], timeout=5).getcode()
```

Comment:
after looking through the travis errors, I'm thinking we may want to do just a little bit more here to help keep things  readable... but that shouldn't necessarily be blocking on this PR, and I'm happy to throw an issue in the backlog. I'd say, if you have a clear idea of what the major offending failure modes are, no reason not to add it now, otherwise feel free to make an issue and we'll come back to it later. sound like a plan?

Reply:
yes, that makes sense. I will investigate that. Would be nice to see the  http response code in the report overview ;-)

Reply:
I added a pytest plugin called [tldr](https://github.com/ismir/mir-datasets/pull/8/commits/9f19d5886eca9751e51d775d58b3ae14b8533ed7) that improves readability. Also I split the test function in metadata test and in url availability. Available in 9f19d58.